docs/src/main/java/org/geotools/tutorial/csv3/CSVDataStore.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ * 	  (C) 2014 - 2015 Open Source Geospatial Foundation (OSGeo)
+ * 	  (c) 2012 - 2014 OpenPlans
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.tutorial.csv3;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import org.geotools.data.FeatureReader;
+import org.geotools.data.FeatureWriter;
+import org.geotools.data.FileDataStore;
+import org.geotools.data.Query;
+import org.geotools.data.Transaction;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.data.store.ContentDataStore;
+import org.geotools.data.store.ContentEntry;
+import org.geotools.data.store.ContentFeatureSource;
+import org.geotools.feature.NameImpl;
+import org.geotools.tutorial.csv3.parse.CSVStrategy;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.Filter;
+
+public class CSVDataStore extends ContentDataStore implements FileDataStore {
+
+    private final CSVStrategy csvStrategy;
+
+    private final CSVFileState csvFileState;
+
+    public CSVDataStore(CSVFileState csvFileState, CSVStrategy csvStrategy) {
+        this.csvFileState = csvFileState;
+        this.csvStrategy = csvStrategy;
+    }
+
+    // docs start getTypeName
```

Comment:
I read somewhere about un-tangling the docs from the gt-csv code... if it has happened these comments here (and probably other places) are not needed anymore. But not harmful either.

Reply:
They do no harm but taking them out will probably be good

Reply:
Agreed, the production CSV instance will gather fixes and optimizations over time that can impact its use as a teaching tool